### PR TITLE
Fixing MeanSTDFilter

### PR DESCRIPTION
**Pull request by richardliaw**
This should address #1081. 

-  [ ] Should add tests to make sure this isn't broken in the future

@ericl @pcmoritz @robertnishihara

### CODE REVIEW

File: python/ray/rllib/ppo/filter.py [hunk 1]
```diff
@@ -92,16 +92,21 @@ def __init__(self, shape, demean=True, destd=True, clip=10.0):
         self.destd = destd
         self.clip = clip
         self.rs = RunningStat(shape)
+        self.buffer = RunningStat(shape)
+
+    def clear_buffer(self):
```
Comment (robertnishihara) [inline python/ray/rllib/ppo/filter.py]: Is this called anywhere?

Comment (robertnishihara) [reply python/ray/rllib/ppo/filter.py]: I see, we don't need it for PPO because we always just create a new RunningStat object for each rollout, right?

**Comment by richardliaw** [comment]
Yeah, maybe should document that...

File: python/ray/rllib/ppo/filter.py [hunk 2]
```diff
@@ -92,16 +92,21 @@ def __init__(self, shape, demean=True, destd=True, clip=10.0):
         self.destd = destd
         self.clip = clip
         self.rs = RunningStat(shape)
+        self.buffer = RunningStat(shape)
```
Comment (robertnishihara) [inline python/ray/rllib/ppo/filter.py]: I'd suggest adding a doc explaining the purpose of `self.buffer` and how it is used in `self.update`. That's the main thing that could be confusing.

**Comment by robertnishihara** [comment]
Looks good to me, we can merge it if you're happy with it. If you want to add tests, we could make sure to run this code path in Jenkins (or perhaps switch it back to the default observation filter).

Did you verify that it fixes #1081?

**Comment by richardliaw** [comment]
Yeah, the example script runs correctly now.

**Review by robertnishihara — APPROVED** [review]

**Pull request merged**